### ISSUE TITLE
How to load an unlabeled dataset

### ISSUE THREAD

**Issue by zhangboshi** [hnuzhy/SSDA-YOLO#3]
![捕获](https://user-images.githubusercontent.com/59415080/204119497-b4c7db25-b20c-41bf-824d-a6dce3f8bfe4.PNG)
I would like to inquire about the problem. Why is there an error that the target data set does not have labels during training?

**Comment by hnuzhy** [comment]
> ![捕获](https://user-images.githubusercontent.com/59415080/204119497-b4c7db25-b20c-41bf-824d-a6dce3f8bfe4.PNG) I would like to inquire about the problem. Why is there an error that the target data set does not have labels during training?


It looks like that you are training your custom dataset instead of PascalVOC2Clipart or CitySpaces2CitySpacesFoggy. If so, you should organize your dataset labels following the YOLOv5 format. Plesase refer scripts under https://github.com/hnuzhy/SSDA-YOLO/tree/master/data/formats/. Also, if you want reproduce my DAOD method, you should remember prepare five different sub-datasets as described in https://github.com/hnuzhy/SSDA-YOLO#yamls

**Comment by zhangboshi** [comment]
> > ![捕获](https://user-images.githubusercontent.com/59415080/204119497-b4c7db25-b20c-41bf-824d-a6dce3f8bfe4.PNG) I would like to inquire about the problem. Why is there an error that the target data set does not have labels during training?
> 
> It looks like that you are training your custom dataset instead of PascalVOC2Clipart or CitySpaces2CitySpacesFoggy. If so, you should organize your dataset labels following the YOLOv5 format. Plesase refer scripts under https://github.com/hnuzhy/SSDA-YOLO/tree/master/data/formats/. Also, if you want reproduce my DAOD method, you should remember prepare five different sub-datasets as described in https://github.com/hnuzhy/SSDA-YOLO#yamls

Thanks for your feedback and guidance. I read your paper and in the training configuration it describes that the training images contain (Is,Isf) with labels and (It,Itf) without labels. But when I load the image of (It, Itf), at the red dot mark in the code position of the screenshot below, an error appears in the above screenshot. With respect to the dataset of (It,Itf), do I need to generate pseudo-labels ahead of time?
![11](https://user-images.githubusercontent.com/59415080/204121559-9f53c756-2cfe-486b-8908-ec8797ae2f45.PNG)

**Comment by hnuzhy** [comment]
> > > ![捕获](https://user-images.githubusercontent.com/59415080/204119497-b4c7db25-b20c-41bf-824d-a6dce3f8bfe4.PNG) I would like to inquire about the problem. Why is there an error that the target data set does not have labels during training?
> > 
> > 
> > It looks like that you are training your custom dataset instead of PascalVOC2Clipart or CitySpaces2CitySpacesFoggy. If so, you should organize your dataset labels following the YOLOv5 format. Plesase refer scripts under https://github.com/hnuzhy/SSDA-YOLO/tree/master/data/formats/. Also, if you want reproduce my DAOD method, you should remember prepare five different sub-datasets as described in https://github.com/hnuzhy/SSDA-YOLO#yamls
> 
> Thanks for your feedback and guidance. I read your paper and in the training configuration it describes that the training images contain (Is,Isf) with labels and (It,Itf) without labels. But when I load the image of (It, Itf), at the red dot mark in the code position of the screenshot below, an error appears in the above screenshot. With respect to the dataset of (It,Itf), do I need to generate pseudo-labels ahead of time? ![11](https://user-images.githubusercontent.com/59415080/204121559-9f53c756-2cfe-486b-8908-ec8797ae2f45.PNG)

Yes. Although we do not use GT labels of datasets from the target domain, we also need to provide corresponding labels for Oracle training (direct transfer without domain adaptation) and Adaptive Vlidation on the test-set of target domain. Please refer code lines 441~534 in `ssda_yolov5_train.py` for more details. Pay special attention to the variable GT labels `targets_tr`, which we do not use during training. The rough pseudo-labels of `targets_tr` saved in variable `pred_tf_nms` are generated online in code line 501.

**Comment by zhangboshi** [comment]
> > > > ![捕获](https://user-images.githubusercontent.com/59415080/204119497-b4c7db25-b20c-41bf-824d-a6dce3f8bfe4.PNG) I would like to inquire about the problem. Why is there an error that the target data set does not have labels during training?
> > > 
> > > 
> > > It looks like that you are training your custom dataset instead of PascalVOC2Clipart or CitySpaces2CitySpacesFoggy. If so, you should organize your dataset labels following the YOLOv5 format. Plesase refer scripts under https://github.com/hnuzhy/SSDA-YOLO/tree/master/data/formats/. Also, if you want reproduce my DAOD method, you should remember prepare five different sub-datasets as described in https://github.com/hnuzhy/SSDA-YOLO#yamls
> > 
> > 
> > Thanks for your feedback and guidance. I read your paper and in the training configuration it describes that the training images contain (Is,Isf) with labels and (It,Itf) without labels. But when I load the image of (It, Itf), at the red dot mark in the code position of the screenshot below, an error appears in the above screenshot. With respect to the dataset of (It,Itf), do I need to generate pseudo-labels ahead of time? ![11](https://user-images.githubusercontent.com/59415080/204121559-9f53c756-2cfe-486b-8908-ec8797ae2f45.PNG)
> 
> Yes. Although we do not use GT labels of datasets from the target domain, we also need to provide corresponding labels for Oracle training (direct transfer without domain adaptation) and Adaptive Vlidation on the test-set of target domain. Please refer code lines 441~534 in `ssda_yolov5_train.py` for more details. Pay special attention to the variable GT labels `targets_tr`, which we do not use during training. The rough pseudo-labels of `targets_tr` saved in variable `pred_tf_nms` are generated online in code line 501.

Thanks again for your guidance. I successfully got it working on a custom dataset.

**Comment by chos1npc** [comment]
I encountered the same problem. I have the test set labels and have placed them in the appropriate folder, but the program still prompts me to provide the train.cache file. How can I solve this issue?

**Comment by DuyunliangToon** [comment]
Hi, I also have the same problem. In the configuration file of the code, train_source_real and train_source_fake need labels, and train_target_real and train_target_fake do not need labels, but I am prompted to need the label of train_target_real during training. In the label cache I observed that the labels of train_source_fake are not used. Do you have a solution?
